walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,121 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+
+
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+      Collection<String> attendees = request.getAttendees();
+      Collection<String> optional = request.getOptionalAttendees();
+      long duration = request.getDuration();
+      //EDGE CASES
+      //return whole day if the meeting to schedule requires no attendees
+      if (attendees.size() == 0 && optional.size() == 0) {
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+      //return no availability if the meeting to schedule is longer than the day
+      else if (duration > 24*60) {
```

Comment:
`if` should be good enough since you `return` in every condition.

Reply:
Done, and fixed corresponding indentation issues.